### PR TITLE
Spring intro

### PR DESCRIPTION
**Pull request by Ketch92**

### CODE REVIEW

**Review by vlykhvar — APPROVED** [review]

File: src/main/java/core/dao/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserDaoImpl implements UserDao {
+    @Autowired
+    private SessionFactory sessionFactory;
```
Comment (Sofasmile) [inline src/main/java/core/dao/UserDaoImpl.java]: https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_01.html
> What is better: setter, constructor, or field injection? Think why.

File: src/main/java/core/service/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import core.dao.UserDao;
+import core.model.User;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Autowired
```
Comment (Sofasmile) [inline src/main/java/core/service/UserServiceImpl.java]: do we need to use Autowired here?

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/core/dao/UserDaoImpl.java (hunk 1 above)
Comment (Ketch92) [reply src/main/java/core/dao/UserDaoImpl.java]: Changed to Constructor injection

File: src/main/java/core/service/UserServiceImpl.java (hunk 2 above)
Comment (Ketch92) [reply src/main/java/core/service/UserServiceImpl.java]: Just forgot to remove it, sorry

**Review by Sofasmile — CHANGES_REQUESTED** [review]
travis failed

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**